solutions/swb-reference/src/dynamicRouteConfig.ts

```diff
@@ -0,0 +1,533 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DynamicRoutesMap, RoutesIgnored } from '@aws/workbench-core-authorization';
+import { SwbAuthZSubject } from './constants';
+
+export const dynamicRoutesMap: DynamicRoutesMap = {
+  '/awsAccounts': {
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_AWS_ACCOUNT,
+          subjectId: '*'
+        }
+      }
+    ],
+    POST: [
+      {
+        action: 'CREATE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_AWS_ACCOUNT,
+          subjectId: '*'
+        }
+      }
+    ]
+  },
+  '/awsAccountTemplateUrls': {
+    POST: [
+      {
+        action: 'CREATE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_AWS_ACCOUNT_TEMPLATE_URL,
+          subjectId: '*'
+        }
+      }
+    ]
+  },
+  '/awsAccounts/:awsAccountId': {
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_AWS_ACCOUNT,
+          subjectId: '${awsAccountId}'
+        }
+      }
+    ],
+    PATCH: [
+      {
+        action: 'UPDATE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_AWS_ACCOUNT,
+          subjectId: '${awsAccountId}'
+        }
+      }
+    ]
+  },
+  '/costCenters': {
+    POST: [
+      {
+        action: 'CREATE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_COST_CENTER,
+          subjectId: '*'
+        }
+      }
+    ],
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_COST_CENTER,
+          subjectId: '*'
+        }
+      }
+    ]
+  },
+  '/costCenters/:costCenter': {
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_COST_CENTER,
+          subjectId: '${costCenter}'
+        }
+      }
+    ],
+    PATCH: [
+      {
+        action: 'UPDATE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_COST_CENTER,
+          subjectId: '${costCenter}'
+        }
+      }
+    ]
+  },
+  '/costCenters/:costCenter/softDelete': {
+    PUT: [
+      {
+        action: 'DELETE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_COST_CENTER,
+          subjectId: '${costCenter}'
+        }
+      }
+    ]
+  },
+  '/datasets': {
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_DATASET,
+          subjectId: '*'
```

Comment:
should this be limited in scope? This allowing reading of all datasets

Reply:
These will need to get refactored when @aws-tyler updates datasets.